### PR TITLE
[DECO-279] Add Prometheus Operator stack

### PR DESCRIPTION
**Pull request by austince**
Adds Prometheus Operator support via a [`ServiceMonitor`](https://github.com/prometheus-operator/prometheus-operator/blob/master/Documentation/api.md#servicemonitor) for all Flink jobs. I've tested both the stacks to ensure the current method still works as well.

Also:
- [hotfix] Use Flink 1.12 for the deployment

### CODE REVIEW

**Comment by austince** [comment]
@knaufk @qinjunjerry optional for review, will solicit one from the deco team as well.

**Review by knaufk — CHANGES_REQUESTED** [review]
Thank you for this addition.

I am wondering if we we need the complexity of both Prometheus deployment modes here? Could we only offer the Operator-based approach in the playground going forward?

I verified that in both modes the metrics are scraped and the Grafana Dashboard looks good. The link templating based on the `deploymentName` does not work though. This change would need to be reverted, I think.

File: values-vvp-add-metrics.yaml [hunk 1]
```diff
@@ -1,4 +1,4 @@
 vvp:
   ui:
     linkTemplates:
-      metrics: http://localhost:3000/d/77Nra5XZz/flink-dashboard?var-deploymentId=<%= deploymentId %>
+      metrics: http://localhost:3000/d/77Nra5XZz/flink-dashboard?var-deploymentName=<%= deploymentName %>
```
[the remark this answers is not in the document]
Comment (qinjunjerry) [reply values-vvp-add-metrics.yaml:4]: Indeed, and we have a JIRA requesting `deploymentName` as a placeholder: FEED-295``

**Review by qinjunjerry — COMMENTED** [review]
As what Konstantin mentioned, scraping works correctly. But the Metrics URL is not working anymore because `deploymentName` is not a valid placeholder.  And because of this, Safari does not show the button row (where the button Start/Suspend/Cancel/.... is) anymore on the deployment detail page, although Chrome & Firefox still shows.

File: values-vvp-add-metrics.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (austince) [reply values-vvp-add-metrics.yaml:4]: Ah hmm :/ ok I'll revert this change, unfortunately.

**Comment by austince** [comment]
> I am wondering if we we need the complexity of both Prometheus deployment modes here? Could we only offer the Operator-based approach in the playground going forward?

I think that would work, though the Prometheus users are still split on installation approaches – not everyone uses the Prometheus Operator. I guess one solution might be to add the annotations to make it work with a "standalone" Prometheus setup, so that it will still work for those users, and just use the Prometheus Operator stack here.

**Comment by austince** [comment]
From feedback from @qinjunjerry, will use the Service + ServiceMonitor approach which will work with any VVP 2.x series. Requires more k8s configuration, but less VVP configuration.

**Review by XComp — COMMENTED** [review]
That is unrelated to this change, but anyway: I started `./setup.sh --with-metrics --with-logging` after running `./setup.sh --edition community`
* I ran into the `ERROR: No Ververica Platform license provided.` in both cases. I assume that for the first one the user shouldn't have to deal with it considering that `--edition community` was specified?
* `> Waiting for all Deployments and Pods to become ready...` took quite a bit of time in the second setup run. It might be nice to have some feedback for the user (process bar or "x out of y pods are running" message)

**Comment by qinjunjerry** [comment]
> That is unrelated to this change, but anyway: I started `./setup.sh --with-metrics --with-logging` after running `./setup.sh --edition community`
> * I ran into the `ERROR: No Ververica Platform license provided.` in both cases. I assume that for the first one the user shouldn't have to deal with it considering that `--edition community` was specified?
`./setup.sh --with-metrics --with-logging --edition enterprise` needs a valid license. `--edition community` needs you to accept the community license.

**Comment by XComp** [comment]
I see, confirming the community license makes sense. But the error printout is kind of not expected, I guess:
```
Error: template: ververica-platform/templates/pvc.yaml:1:11: executing "ververica-platform/templates/pvc.yaml" at <include "vvp.should.create.pvc" .>: error calling include: template: ververica-platform/templates/_helpers.tpl:75:15: executing "vvp.should.create.pvc" at <include "vvp.pvc.enabled" .>: error calling include: template: ververica-platform/templates/_
helpers.tpl:67:12: executing "vvp.pvc.enabled" at <include "vvp.private.process.application.config" .>: error calling include: template: ververica-platform/templates/_helpers.tpl:137:8: executing "vvp.private.process.application.config" at <include "vvp.private.inject.license.config" .>: error calling include: template: ververica-platform/templates/_helpers.tpl:
212:106: executing "vvp.private.inject.license.config" at <fail>: error calling fail:

================================================================================
ERROR: No Ververica Platform license provided.

Actions to resolve:

* Provide a valid license at `vvp.license.data`.

- OR -

* Read the printed Ververica Platform Community Edition license agreement and
indicate that you accept it by setting `acceptCommunityEditionLicense=true`.
```
That was part of the output when calling `./setup.sh --edition community`

**Comment by qinjunjerry** [comment]
Yes, the error is printed there. I never noticed it and simply type `y' to the question following the long LICENSE AGREEMENT
```
...
This Agreement constitutes the parties’ entire understanding regarding the Software, and supersedes any and all other prior or contemporaneous agreements, whether written or oral.


================================================================================
ERROR: No Ververica Platform license provided.

Actions to resolve:

* Provide a valid license at `vvp.license.data`.

- OR -

* Read the printed Ververica Platform Community Edition license agreement and
indicate that you accept it by setting `acceptCommunityEditionLicense=true`.
================================================================================
Do you want to pass 'acceptCommunityEditionLicense=true'? (y/N) y
```

**Comment by austince** [comment]
I do not have push access to this branch anymore, so will update tomorrow.

**Review by qinjunjerry — APPROVED** [review]
Thanks Austin for providing the example! The new commit looks good to me.

**Review by knaufk — APPROVED** [review]
LGTM

**Review by XComp — APPROVED** [review]
LGTM

**Pull request merged**